test/e2e/e2e.go

```diff
@@ -175,6 +191,64 @@ var _ = Describe("validation", func() {
 			})
 		})
 	})
+
+	Context("Creating AddressPool", func() {
+		It("Should create addresspool CR successfully", func() {
+			By("By creating a configmap")
+			addresspool := &metallbv1alpha.AddressPool{
```

Comment:
I think we should continue loading the files in config/samples (if we already went that way)

Reply:
I don't think using samples is a good way for CI, normally sample is to share config sample with users that is why I took that route and I have seem other operators do the same

Reply:
we went that route because samples find themselves often in production :) adding @fedepaol as he and I also talked about it previously

Reply:
indeed samples should accurate and reflect possible configs but not seen anywhere the samples are also used by CI

Reply:
If the examples are the same as we use here, then validating them could help.

In any case, I'd try to add more coverage converting this into a table driven test (https://onsi.github.io/ginkgo/#table-driven-tests) , passing in the addressPool object and the configMap you expect to find.
One degree is the addition of autoassign flag.

Reply:
the sample config is different and will diverge as more testing with metallb is done mostly exploring possible configs with metallb not related to the operator directly, I will defer the table since no more CR for this release

Reply:
The table is not for different CRs, but for different versions of the same CR.

What I am talking about is having different addresspools mapped to different configmaps and verifying that the rendering is what we expect. We want to ensure that the rendering is done properly, not that metallb behaves correctly according to a given config.